libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/client/EvcsClient.java

```diff
@@ -0,0 +1,191 @@
+            throws EvcsServiceException {
+        try {
+            HttpRequest.Builder httpRequestBuilder =
+                    HttpRequest.newBuilder()
+                            .uri(getUri(userId, null))
+                            .POST(
+                                    HttpRequest.BodyPublishers.ofString(
+                                            OBJECT_MAPPER.writeValueAsString(userVCsForEvcs)));
+            httpRequestBuilder.header(
+                    X_API_KEY_HEADER, configService.getAppApiKey(EVCS_APP_ID.getPath()));
+            httpRequestBuilder.header(CONTENT_TYPE, APPLICATION_JSON_CHARSET_UTF_8);
+
+            var evcsHttpResponse = sendHttpRequest(httpRequestBuilder.build());
+            checkResponseStatusCode(evcsHttpResponse);
+        } catch (URISyntaxException e) {
+            throw new EvcsServiceException(
+                    HTTPResponse.SC_SERVER_ERROR, ErrorResponse.FAILED_TO_CONSTRUCT_EVCS_URI);
+        } catch (JsonProcessingException e) {
+            throw new EvcsServiceException(
+                    HTTPResponse.SC_SERVER_ERROR, ErrorResponse.FAILED_TO_PARSE_EVCS_REQUEST_BODY);
+        }
+    }
+
+    @Tracing
+    public void updateEvcsUserVCs(String userId, List<EvcsUpdateUserVCsDto> evcsUserVCsToUpdate)
+            throws EvcsServiceException {
+        try {
+            HttpRequest.Builder httpRequestBuilder =
+                    HttpRequest.newBuilder()
+                            .uri(getUri(userId, null))
+                            .method(
+                                    "PATCH",
+                                    HttpRequest.BodyPublishers.ofString(
+                                            OBJECT_MAPPER.writeValueAsString(evcsUserVCsToUpdate)));
+            httpRequestBuilder.header(
+                    X_API_KEY_HEADER, configService.getAppApiKey(EVCS_APP_ID.getPath()));
+            httpRequestBuilder.header(CONTENT_TYPE, APPLICATION_JSON_CHARSET_UTF_8);
+
+            var evcsHttpResponse = sendHttpRequest(httpRequestBuilder.build());
+            checkResponseStatusCode(evcsHttpResponse);
+        } catch (URISyntaxException e) {
+            throw new EvcsServiceException(
+                    HTTPResponse.SC_SERVER_ERROR, ErrorResponse.FAILED_TO_CONSTRUCT_EVCS_URI);
+        } catch (JsonProcessingException e) {
+            throw new EvcsServiceException(
+                    HTTPResponse.SC_SERVER_ERROR, ErrorResponse.FAILED_TO_PARSE_EVCS_REQUEST_BODY);
+        }
+    }
+
+    private URI getUri(String userId, List<EvcsVCState> vcStatesToQueryFor)
+            throws URISyntaxException {
+        if (vcStatesToQueryFor != null) {
+            return new URIBuilder(
+                            configService
+                                    .getSsmParameter(EVCS_APPLICATION_URL)
+                                    .concat("/")
+                                    .concat(userId))
```

Comment:
```suggestion
            return new URIBuilder(
                            configService
                                    .getSsmParameter(EVCS_APPLICATION_URL))
                            .setPath(userId)
```

Reply:
I'm also wondering if we should be URL encoding the userId. It will contain a bunch of colons which could be confusing. I suppose it depends what the actual EVCS is expecting to receive though...

Reply:
URIBuilder take care of encoding userId - tried one scenario working fine.